odl/tomo/geometry/detector.py

```diff
@@ -0,0 +1,459 @@
+        """The partial derivative(s) of the surface parametrization.
+
+        Parameters
+        ----------
+        param : element of `params`
+            The parameter value where to evaluate the function
+        """
+        raise NotImplementedError
+
+    def surface_measure(self, param):
+        """The density function of the surface measure.
+
+        This is the default implementation relying on the `surface_deriv`
+        method. For ``ndim == 1``, the density is given by the `Arc
+        length`_, for ``ndim == 2``, it is the length of the cross product
+        of the partial derivatives of the parametrization, see Wikipedia's
+        `Surface area`_ article.
+
+        Parameters
+        ----------
+        param : element of `params`
+            The parameter value where to evaluate the function
+
+        Returns
+        -------
+        measure : `float`
+            The density value at the given parameter
+
+        .. _Arc length:
+            https://en.wikipedia.org/wiki/Curve#Lengths_of_curves
+        .. _Surface area:
+            https://en.wikipedia.org/wiki/Surface_area
+        """
+        if param not in self.params:
+            raise ValueError('parameter value {} not in the valid range {}.'
+                             ''.format(param, self.params))
+        if self.ndim == 1:
+            return float(np.linalg.norm(self.surface_deriv(param)))
+        elif self.ndim == 2:
+            return float(np.linalg.norm(np.cross(*self.surface_deriv(param))))
+        else:
+            raise NotImplementedError
+
+
+class FlatDetector(with_metaclass(ABCMeta, Detector)):
+
+    """Abstract class for flat detectors in 2 and 3 dimensions."""
+
+    def surface_measure(self, param):
+        """The constant density function of the surface measure.
+
+        Parameters
+        ----------
+        param : element of `params`
+            The parameter value where to evaluate the function
+
+        Returns
+        -------
+        meas : `float`
+            The constant density 1.0
+        """
+        if param not in self.params:
+            raise ValueError('parameter value {} not in the valid range '
+                             '{}.'.format(param, self.params))
+        return 1.0
+
+    def __repr__(self):
+        """d.__repr__() <==> repr(d)."""
+        inner_fstr = '{!r}'
+        if self.has_sampling:
+            inner_fstr += ',\n grid={grid!r}'
+        inner_str = inner_fstr.format(self.params, grid=self.param_grid)
+        return '{}({})'.format(self.__class__.__name__, inner_str)
+
+    def __str__(self):
+        """d.__str__() <==> str(d)."""
+        # TODO: prettify
+        inner_fstr = '{}'
+        if self.has_sampling:
+            inner_fstr += ',\n grid={grid}'
+        inner_str = inner_fstr.format(self.params, grid=self.param_grid)
+        return '{}({})'.format(self.__class__.__name__, inner_str)
+
+
+# TODO: rename to Flat1dDetector to be consitent with Flat2dDetector? or
```

Comment:
Since I also prefer `Flat2dDetector`, I renamed `LineDetector` to `Flat1dDetector`.